link/src/main/java/com/stripe/android/link/ui/verification/VerificationViewModel.kt

```diff
@@ -0,0 +1,220 @@
+            otpCode.collect { code ->
+                code?.let { onVerificationCodeEntered(code) }
+            }
+        }
+    }
+
+    fun onVerificationCodeEntered(code: String) {
+        updateViewState {
+            it.copy(
+                isProcessing = true,
+                errorMessage = null,
+            )
+        }
+
+        viewModelScope.launch {
+            linkAccountManager.confirmVerification(code).fold(
+                onSuccess = {
+                    updateViewState {
+                        it.copy(isProcessing = false)
+                    }
+
+                    linkEventsReporter.on2FAComplete()
+                    navigateAndClearStack(LinkScreen.Wallet)
+                },
+                onFailure = {
+                    linkEventsReporter.on2FAFailure()
+                    for (i in 0 until otpElement.controller.otpLength) {
+                        otpElement.controller.onValueChanged(i, "")
+                    }
+                    onError(it)
+                }
+            )
+        }
+    }
+
+    fun startVerification() {
+        updateViewState {
+            it.copy(errorMessage = null)
+        }
+
+        viewModelScope.launch {
+            val result = linkAccountManager.startVerification()
+            val error = result.exceptionOrNull()
+
+            updateViewState {
+                it.copy(
+                    isSendingNewCode = false,
+                    didSendNewCode = error == null,
+                    errorMessage = error?.getErrorMessage()?.resolvableString,
+                )
+            }
+        }
+    }
+
+    fun resendCode() {
+        updateViewState { it.copy(isSendingNewCode = true) }
+        startVerification()
+    }
+
+    fun didShowCodeSentNotification() {
+        updateViewState {
+            it.copy(didSendNewCode = false)
+        }
+    }
+
+    fun onBack() {
+        clearError()
+        goBack()
+        linkEventsReporter.on2FACancel()
+        viewModelScope.launch {
+            linkAccountManager.logOut()
+        }
+    }
+
+    fun onChangeEmailClicked() {
+        clearError()
+        navigateAndClearStack(LinkScreen.SignUp)
+        viewModelScope.launch {
+            linkAccountManager.logOut()
+        }
+    }
+
+    fun onFocusRequested() {
+        updateViewState {
+            it.copy(requestFocus = false)
+        }
+    }
+
+    private fun clearError() {
+        updateViewState {
+            it.copy(errorMessage = null)
+        }
+    }
+
+    private fun onError(error: Throwable) = error.getErrorMessage().let { message ->
+        logger.error("VerificationViewModel Error: ", error)
+
+        updateViewState {
+            it.copy(
+                isProcessing = false,
+                errorMessage = when (message) {
+                    is ErrorMessage.FromResources -> message.stringResId.resolvableString
+                    is ErrorMessage.Raw -> message.errorMessage.resolvableString
+                },
```

Comment:
can you use the ErrorMessage.resolvableString extension function defined below for this?